common/lib/dependabot/pull_request_creator/labeler.rb

```diff
@@ -174,7 +174,10 @@ def create_default_language_label_if_required
       end
 
       def default_labels_for_pr
-        if custom_labels then custom_labels & labels
+        if custom_labels
+          # Azure does not have centralised labels
+          custom_labels if source.provider == "azure"
```

Comment:
Did you try this after the last push to fix CI? I think with the current logic the above line is going be ignored. It should be 

```suggestion
          return custom_labels if source.provider == "azure"
```

If the above is correct, we should probably add a spec at `common/spec/dependabot/pull_request_creator/labeler_spec.rb` that catches the bug.